### ISSUE TITLE
systemd-networkd coredumps for DNS-over-TLS with openssl 1.1.1c

### ISSUE THREAD

**Issue by Keruspe** [systemd/systemd#12781]
**systemd version the issue has been seen with**
> 242

**Used distribution**
 > Exherbo

**Expected behaviour you didn't see**
 > systemd-networkd starts and works

**Unexpected behaviour you saw**
 > coredump

```
                #0  0x00007f545bcb3e93 __libc_free (libc.so.6)
                #1  0x00007f545bf7f04f CRYPTO_clear_realloc (libcrypto.so.1.1)
                #2  0x00007f545bee18b9 BUF_MEM_grow_clean (libcrypto.so.1.1)
                #3  0x00007f545bec4104 mem_write (libcrypto.so.1.1)
                #4  0x00007f545bebfbfa bwrite_conv (libcrypto.so.1.1)
                #5  0x00007f545bebed23 bio_write_intern (libcrypto.so.1.1)
                #6  0x00007f545bebf197 BIO_write (libcrypto.so.1.1)
                #7  0x00007f545c0f3de5 ssl3_write_pending (libssl.so.1.1)
                #8  0x00007f545c0f4cd7 do_ssl3_write.localalias.3 (libssl.so.1.1)
                #9  0x00007f545c0f4f06 ssl3_write_bytes (libssl.so.1.1)
                #10 0x00007f545c1072f3 ssl_write_internal (libssl.so.1.1)
                #11 0x00007f545c107405 SSL_write (libssl.so.1.1)
                #12 0x00005652c23e201c dnstls_stream_write (systemd-resolved)
                #13 0x00005652c23db8e7 dns_stream_writev (systemd-resolved)
                #14 0x00005652c23dc41c on_stream_io (systemd-resolved)
                #15 0x00007f545c411700 source_dispatch (libsystemd-shared-242.so)
                #16 0x00007f545c413410 sd_event_dispatch (libsystemd-shared-242.so)
                #17 0x00007f545c4135c8 sd_event_run (libsystemd-shared-242.so)
                #18 0x00007f545c4137fb sd_event_loop (libsystemd-shared-242.so)
                #19 0x00005652c23b2911 run (systemd-resolved)
                #20 0x00007f545bc52c03 __libc_start_main (libc.so.6)
                #21 0x00005652c23b2b3e _start (systemd-resolved)
```

**Steps to reproduce the problem**

Update openssl from 1.1.1b to 1.1.1c. It used to work just fine with 1.1.1b

**Comment by yuwata** [comment]
I hope it is already fixed by 18bddea.

**Comment by Keruspe** [comment]
Thanks for the pointer, seems to work indeed
